platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultDeprecationData.java

```diff
@@ -31,6 +33,20 @@ public Type getType() {
         return type;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Is this only about consistency, or do we actually need to depend on the equals implementaion?

Reply:
I started this because I was thinking about addressing [#32226](https://github.com/gradle/gradle/issues/32226).
I think to address that we'll need working `equals` and `hashcode`.